admin/src/core/services/cognito.service.ts

```diff
@@ -0,0 +1,214 @@
+    const url = window.location.href;
+    name = name.replace(/[\[\]]/g, "\\$&");
+    const regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)");
+    const results = regex.exec(url);
+    if (!results) {
+      return null;
+    }
+    if (!results[2]) {
+      return "";
+    }
+    return decodeURIComponent(results[2].replace(/\+/g, " "));
+  }
+
+  /**
+   * See OIDC Attribute Mapping mapping reference:
+   *      https://github.com/bcgov/nr-forests-access-management/wiki/OIDC-Attribute-Mapping
+   * Note, current user data return for 'userData.username' is matched to "cognito:username" on Cognito.
+   * Which isn't what we really want to display. The display username is "custom:idp_username" from token.
+   */
+  private parseToken(authToken: CognitoUserSession): object {
+    const decodedIdToken = authToken.getIdToken().decodePayload();
+    const decodedAccessToken = authToken.getAccessToken().decodePayload();
+    return {
+      id_token: decodedIdToken,
+      access_token: decodedAccessToken,
+    };
+  }
+
+  public async init(): Promise<any> {
+    /*
+        See Aws-Amplify documenation: 
+        https://docs.amplify.aws/lib/auth/social/q/platform/js/
+        https://docs.amplify.aws/lib/auth/advanced/q/platform/js/#identity-pool-federation
+    */
+
+    let url: string =
+      this.configService.getApiBasePath() + "/api/awsCognitoConfig";
+    let data = await this.http
+      .get(url, { observe: "body", responseType: "json" })
+      .toPromise();
+    this.config = data as AwsCognitoConfig;
+
+    const parsedConfig = {
+      aws_cognito_region: this.config.region,
+      aws_user_pools_id: this.config.userPoolsId,
+      aws_user_pools_web_client_id: this.config.userPoolWebClientId,
+      aws_mandatory_sign_in: this.config.mandatorySignIn ? "enable" : "disable",
+      oauth: {
+        domain: `${this.config.domain}.auth.${this.config.region}.amazoncognito.com`,
+        scope: this.config.scope,
+        redirectSignIn: `${this.config.frontendRedirectBaseUrl}/admin`,
+        redirectSignOut: `${this.config.frontendRedirectBaseUrl}/admin/not-authorized?loggedout=true`,
+        responseType: this.config.signUpVerificationMethod,
+      },
+      federationTarget: this.config.federationTarget,
+    };
+
+    console.log("Using cognito config = " + JSON.stringify(this.config));
+    Amplify.configure(parsedConfig);
+
+    this.loggedOut = this.getParameterByName("loggedout");
+
+    if (!this.config.enabled) {
+      this.fakeUser = getFakeUser();
+      this.initialized = true;
+      return null;
+    }
+
+    return new Promise(async (resolve, reject) => {
+      if (this.loggedOut === "true") {
+        resolve(null);
+      } else {
+        Auth.currentAuthenticatedUser()
+          .then(async (_userData) => {
+            console.log("_userData", _userData);
+            if (!_userData) {
+              await this.login();
+            } else {
+              await this.refreshToken();
```

Comment:
Unclear why we are calling refreshToken here. Didn't we just get the token?

Reply:
Amplify does not provide token at _userData retruns.
It's the 'refreshToken' which asks Amplify lib to get the actual token back after login.